File: commonlib/src/main/java/com/hjl/commonlib/network/interceptor/LogInterceptor.java
package com.hjl.commonlib.network.interceptor;

import android.util.Log;

import java.io.IOException;

import okhttp3.FormBody;
import okhttp3.Interceptor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.ResponseBody;

/**
 * created by long on 2019/10/21
 */
public class LogInterceptor implements Interceptor {

    public static String TAG = "HTTP LogInterceptor";

    @Override
    public Response intercept(Chain chain) throws IOException {

        Request request = chain.request();
        long startTime = System.currentTimeMillis();
        Response response = chain.proceed(request);
        long endTime = System.currentTimeMillis();
        long duration = endTime - startTime;

        MediaType mediaType = response.body().contentType();
        String content = response.body().string();

        Log.w(TAG,"----------Start----------------");
        Log.i(TAG,"|" + request.toString());
        String method = request.method();
        if ("POST".equals(method)){

            if (request.body() instanceof FormBody){
                FormBody body = (FormBody) request.body();
                for (int i = 0; i < body.size();i++){
                   // sb.append(body.encodedName(i) + " = " + body.encodedValue(i) + ",");
                    Log.i(TAG,"| RequestParams:{ params : "  + body.encodedName(i) + " values: " + body.encodedValue(i) + " }");
                }


            }
        }

        Log.w(TAG, "Response: " );
        Log.e(TAG,"| " + request.toString() + request.headers().toString());
        Log.d(TAG,"| Response:" + content);
        Log.w(TAG,"----------End:" + duration + "毫秒----------");

        return response.newBuilder().body(ResponseBody.create(mediaType,content)).build();
    }

}

File: commonlib/src/main/java/com/hjl/commonlib/network/interceptor/LogInterceptor.kt
package com.hjl.commonlib.network.interceptor

import android.util.Log
import okhttp3.FormBody
import okhttp3.Interceptor
import okhttp3.Response
import okhttp3.ResponseBody.Companion.toResponseBody
import java.io.IOException

/**
 * created by long on 2019/10/21
 */
class LogInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val startTime = System.currentTimeMillis()
        val response = chain.proceed(request)
        val endTime = System.currentTimeMillis()
        val duration = endTime - startTime
        val responseBody = response.body ?: return response
        val mediaType = responseBody.contentType()
        val content = responseBody.string()
        Log.d(TAG, "---------------------Request Start---------------------------")
//        Log.d(TAG, "|${request.url()}")
        Log.d(TAG, "|${request}")
        val method = request.method
        if ("POST" == method) {
            val requestBody = request.body
            if (requestBody is FormBody) {
                for (i in 0 until requestBody.size) {
                    Log.i(
                        TAG,
                        "| RequestParams:{params :${requestBody.encodedName(i)} values:${
                            requestBody.encodedValue(i)
                        }}"
                    )
                }
            }
        }
        Log.d(TAG, "Response: ")
        Log.d(TAG, "| Response:$content")
        Log.d(TAG, "------------------End Request Cost: $duration ms ------------------")
        return response.newBuilder().body(content.toResponseBody(mediaType)).build()
    }

    companion object {
        var TAG = "LogInterceptor"
    }
}
